sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py

```diff
@@ -357,8 +357,7 @@ def _from_rest_object_to_init_params(cls, obj: dict) -> Dict:
                 obj["task"].environment = task_env[len(ARM_ID_PREFIX) :]
 
         if "resources" in obj and obj["resources"]:
-            resources = RestJobResourceConfiguration.from_dict(obj["resources"])
-            obj["resources"] = JobResourceConfiguration._from_rest_object(resources)
+            obj["resources"] = JobResourceConfiguration._from_dict(obj["resources"])
```

Comment:
why need a new function here?

Reply:
the RestJobResourceConfiguration has attribute type checker, the node_count should be a data binding string instead of int type here, so here I changed to load JobResourceConfiguration from dict directly.